packages/main/src/plugin/container-registry.ts

```diff
@@ -193,6 +195,21 @@ export class ContainerProviderRegistry {
     };
     let previousStatus = containerProviderConnection.status();
 
+    providerRegistry.onBeforeDidUpdateContainerConnection(event => {
+      if (event.providerId === provider.id && event.connection.name === containerProviderConnection.name) {
+        const newStatus = event.status;
+        if (newStatus === 'stopped') {
+          internalProvider.api = undefined;
+          internalProvider.libpodApi = undefined;
+          this.apiSender.send('provider-change', {});
+        }
+        if (newStatus === 'started') {
+          this.setupConnectionAPI(internalProvider, containerProviderConnection);
+        }
```

Comment:
before you were using `this.internalProviders.set(id, internalProvider);` should it be kept ?

Reply:
I remove it because I don't think this is needed